server/src/ServerList.cpp

```diff
@@ -876,75 +908,92 @@ void ServerList::msgSVI_REQUESTTEXT(CString& pPacket)
 	CString option = data.readString("\n");
 	CString paramsData = data.readString("");
 
-	auto player = m_server->getPlayer(playerId);
-	if (player != nullptr)
+	if (playerId < 2)
+	{
+		#ifdef V8NPCSERVER
+		// Send event to server that player is logging in
+		for (const auto& [npcName, npcPtr]: m_server->getNPCNameList())
+		{
+			// TODO(joey): check if they have the event before queueing for them
+			if (const auto npcObject = npcPtr.lock(); npcObject && npcName == params[0])
+			{
+				npcObject->queueNpcEvent("npc.receivetext", true, data.toString());
+			}
+		}
```

Comment:
Hm, this is a map on the npc name itself - why the traversal, and why do I need to check if they have the event before queueing for them?!

Reply:
I'll clean this up :)